src/PKSim.Core/Snapshots/Mappers/SnapshotMapper.cs

```diff
@@ -38,18 +49,28 @@ public SnapshotMapper(IRepository<ISnapshotMapperSpecification> snapshotMapperRe
       public object MapToSnapshot(object model)
       {
          var modelType = model.GetType();
-         foreach (var mapper in _allMappers)
-         {
-            if (mapper.IsSatisfiedBy(modelType))
-               return mapper.MapToSnapshot(model);
-         }
-
-         throw new SnapshotNotFoundException(modelType);
+         return mapperFor(modelType).MapToSnapshot(model);
       }
 
       public object MapToModel(object snapshot)
       {
-         throw new NotImplementedException();
+         var snapshotType = snapshot.GetType();
+         return mapperFor(snapshotType).MapToModel(snapshot);
+      }
+
+      public Type SnapshotTypeFor<T>()
+      {
+         var modelType = typeof(T);
+         return mapperFor(modelType).SnapshotTypeFor<T>();
+      }
+
+      private ISnapshotMapper mapperFor(Type modelOrSnapshotType)
+      {
+         var mapper =  _allMappers.FirstOrDefault(x => x.IsSatisfiedBy(modelOrSnapshotType));
```

Comment:
See discussion from previous PR. This is where the non generic interface is required so that this object can old a reference on ALL snapshots mappers